app/code/Mykhailok/SupportChat/Observer/Grid/ChatMessageCollection.php

```diff
@@ -0,0 +1,33 @@
+<?php
+declare(strict_types=1);
+
+namespace Mykhailok\SupportChat\Observer\Grid;
+
+class ChatMessageCollection implements \Magento\Framework\Event\ObserverInterface
+{
+    private \Magento\Framework\App\ResourceConnection $resourceConnection;
+    private \Magento\Framework\App\Request\Http $request;
+
+    /**
+     * ChatMessageCollection constructor.
+     * @param \Magento\Framework\App\ResourceConnection $resourceConnection
+     * @param \Magento\Framework\App\Request\Http $request
+     */
+    public function __construct(
+        \Magento\Framework\App\ResourceConnection $resourceConnection,
+        \Magento\Framework\App\Request\Http $request
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->request = $request;
+    }
+
+    public function execute(\Magento\Framework\Event\Observer $observer)
```

Comment:
There I want to apply the current chat Id to the chat message collection filter.
It works for the my_chat/chat/index action because the request object has the param `chat_id`.
I can not understand how I can take chat id value from the mui/index/render action. In this case, we can use the request param 'referer', but I am not sure that getting and parsing the referrer URL is the simplest way.